cloud/pom.xml

```diff
@@ -0,0 +1,141 @@
+  <modelVersion>4.0.0</modelVersion>
+  <parent>
+    <groupId>org.apache.spark</groupId>
+    <artifactId>spark-parent_2.11</artifactId>
+    <version>2.0.0-SNAPSHOT</version>
+    <relativePath>../pom.xml</relativePath>
+  </parent>
+
+  <artifactId>spark-cloud_2.11</artifactId>
+  <packaging>jar</packaging>
+  <name>Spark Project cloud integration</name>
+  <description>Contains support for cloud infrastructures, including the Hadoop JARs and
+    transitive dependencies needed to interact with the infrastructures.
+    When included in the spark-assembly JAR, the hadoop artifacts are included, but not
+    any of the 3rd party libraries needed, such as those from Amazon (for AWS) and Microsoft (Azure).
+    These will need to be explicitly added to the classpath of any application interacting
+    with the services.
+    
+    Any project which explicitly depends upon the spark-cloud artifact will get the dependencies;
+    the exact versions of which will depend upon the hadoop version Spark was compiled against.
+
+    Hadoop 2.5 and earlier: jets3t.
+    Hadoop 2.6: hadoop-aws and aws-java-sdk JARs
+    Hadoop 2.7+: hadoop-aws, aws-java-sdk-s3, hadoop-azure and azure-storage JARs
+
+    WARNING: the signatures of methods in aws-java-sdk/aws-java-sdk-s3 can change between versions:
+    use the same version against which Hadoop was compiled.
+  </description>
+  <properties>
+    <sbt.project.name>cloud</sbt.project.name>
+  </properties>
+
+  <dependencies>
+    <dependency>
+      <groupId>org.apache.spark</groupId>
+      <artifactId>spark-core_${scala.binary.version}</artifactId>
+      <version>${project.version}</version>
+    </dependency>
+    <dependency>
+      <groupId>org.apache.spark</groupId>
+      <artifactId>spark-core_${scala.binary.version}</artifactId>
+      <version>${project.version}</version>
+      <type>test-jar</type>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.apache.spark</groupId>
+      <artifactId>spark-test-tags_${scala.binary.version}</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>net.java.dev.jets3t</groupId>
+      <artifactId>jets3t</artifactId>
+    </dependency>
+
+    <!-- Explicit listing of transitive deps that are shaded. Otherwise, odd compiler crashes. -->
+    <dependency>
+      <groupId>com.google.guava</groupId>
+      <artifactId>guava</artifactId>
+    </dependency>
+    <!-- End of shaded deps. -->
+  </dependencies>
+  <build>
+    <outputDirectory>target/scala-${scala.binary.version}/classes</outputDirectory>
+    <testOutputDirectory>target/scala-${scala.binary.version}/test-classes</testOutputDirectory>
+  </build>
+
+  <profiles>
+    
+    <!--
+      This profile is enabled automatically by the sbt built. It changes the scope for the guava
```

Comment:
funny: that was a comment I lifted with the dependency cargo cult style; it'll need fixing in in the original too..